src/plan/mutator_context.rs

```diff
@@ -118,3 +140,155 @@ pub trait MutatorContext<VM: VMBinding>: Send + 'static {
         self.barrier().post_write_barrier(WriteTarget::Object(obj));
     }
 }
+
+/// This is used for plans to indicate the number of allocators reserved for the plan.
+/// This is used as a parameter for creating allocator/space mapping.
+/// A plan is required to reserve the first few allocators. For example, if n_bump_pointer is 1,
+/// it means the first bump pointer allocator will be reserved for the plan (and the plan should
+/// initialize its mapping itself), and the spaces in common/base plan will use the following bump
+/// pointer allocators.
+#[allow(dead_code)]
+#[derive(Default)]
+pub(crate) struct ReservedAllocators {
+    pub n_bump_pointer: u8,
+    pub n_large_object: u8,
+    pub n_malloc: u8,
+    pub n_immix: u8,
+    pub n_mark_compact: u8,
+}
+
+impl ReservedAllocators {
+    pub const DEFAULT: Self = ReservedAllocators {
+        n_bump_pointer: 0,
+        n_large_object: 0,
+        n_malloc: 0,
+        n_immix: 0,
+        n_mark_compact: 0,
+    };
+    /// check if the number of each allocator is okay. Panics if any allocator exceeds the max number.
+    fn validate(&self) {
+        use crate::util::alloc::allocators::*;
+        assert!(
+            self.n_bump_pointer as usize <= MAX_BUMP_ALLOCATORS,
+            "Allocator mapping declared more bump pointer allocators than the max allowed."
+        );
+        assert!(
+            self.n_large_object as usize <= MAX_LARGE_OBJECT_ALLOCATORS,
+            "Allocator mapping declared more large object allocators than the max allowed."
+        );
+        assert!(
+            self.n_malloc as usize <= MAX_MALLOC_ALLOCATORS,
+            "Allocator mapping declared more malloc allocators than the max allowed."
+        );
+    }
```

Comment:
Missing assertions for immix and mark compact allocator.

Reply:
Fixed.